Bottom: e0d5eb3f0f271556b1a309117b8ae03ef16f1b4b
Top:    0178788c64d41698eb948088c861d6e10342ea7b
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2014-06-12 13:12:41 -0400

Refresh of 0001-serial-8250_dw-support-DMA-on-the-OF-case.patch

---

diff --git a/drivers/tty/serial/8250/8250_dw.c b/drivers/tty/serial/8250/8250_dw.c
index 1d4f62e..c2f48a5 100644
--- a/drivers/tty/serial/8250/8250_dw.c
+++ b/drivers/tty/serial/8250/8250_dw.c
@@ -258,12 +258,13 @@ static int dw8250_probe_of(struct uart_port *p,
 		}
 	}
 	if (has_ucv)
-		dw8250_setup_port(container_of(p, struct uart_8250_port, port));
+		dw8250_setup_port(up);
 
 	if (!of_property_read_u32(np, "reg-shift", &val))
 		p->regshift = val;
 
-	data->rst = devm_reset_control_get_optional(p->dev, NULL);
+	if (of_get_property(np, "dmas", NULL))
+		up->dma = &data->dma;	data->rst = devm_reset_control_get_optional(p->dev, NULL);
 
 	/* clock got configured through clk api, all done */
 	if (p->uartclk)
